awx_collection/plugins/modules/tower_team.py

```diff
@@ -57,57 +57,68 @@
     tower_config_file: "~/tower_cli.cfg"
 '''
 
-from ..module_utils.ansible_tower import TowerModule, tower_auth_config, tower_check_mode
-
-try:
-    import tower_cli
-    import tower_cli.exceptions as exc
-
-    from tower_cli.conf import settings
-except ImportError:
-    pass
+from ..module_utils.tower_api import TowerModule
 
 
 def main():
-
+    # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
         name=dict(required=True),
+        new_name=dict(required=False),
         description=dict(),
         organization=dict(required=True),
         state=dict(choices=['present', 'absent'], default='present'),
     )
 
+    # Create a module for ourselves
     module = TowerModule(argument_spec=argument_spec, supports_check_mode=True)
 
+    # Extract our parameters
     name = module.params.get('name')
+    new_name = module.params.get('new_name')
     description = module.params.get('description')
     organization = module.params.get('organization')
     state = module.params.get('state')
 
-    json_output = {'team': name, 'state': state}
-
-    tower_auth = tower_auth_config(module)
-    with settings.runtime_values(**tower_auth):
-        tower_check_mode(module)
-        team = tower_cli.get_resource('team')
-
-        try:
-            org_res = tower_cli.get_resource('organization')
-            org = org_res.get(name=organization)
-
-            if state == 'present':
-                result = team.modify(name=name, organization=org['id'],
-                                     description=description, create_on_missing=True)
-                json_output['id'] = result['id']
-            elif state == 'absent':
-                result = team.delete(name=name, organization=org['id'])
-        except (exc.NotFound) as excinfo:
-            module.fail_json(msg='Failed to update team, organization not found: {0}'.format(excinfo), changed=False)
-        except (exc.ConnectionError, exc.BadRequest, exc.NotFound, exc.AuthError) as excinfo:
-            module.fail_json(msg='Failed to update team: {0}'.format(excinfo), changed=False)
-
-    json_output['changed'] = result['changed']
-    module.exit_json(**json_output)
+    # We can either use the default check mode option or we can customize our own
+    module.default_check_mode()
+
+    # Attempt to lookup the org the user specified
+    org_id = module.resolve_name_to_id('organizations', organization)
+
+    # Attempt to lookup team based on the provided name and org ID
+    team = module.get_one('teams', **{
+        'data': {
+            'name': name,
+            'organization': org_id
+        }
+    })
```

Comment:
Observe, how @john-westcott-iv just blows through all of the problems enumerated in https://github.com/ansible/tower-cli/projects/1

This could maybe be consolidated in to some helper utilities... or maybe just don't. Using contextual data to formulate queries for related items is the level of difficulty that it's actually right choice to have a human write those rules, understanding the practical use of the resources.